.github/workflows/rust.yml

```diff
@@ -53,6 +74,16 @@ jobs:
         # the binary will be used by the next cargo test step
         run: cargo build
       - name: Run cargo tests
-        run: cargo test --all-targets --all-features
+        env:
+          GITHUB_ACTIONS_BASH_CONTAINER: 1
+        run: |
+          docker run --volume $PWD:/root --network host --detach --name test_container bash:${{ matrix.bash_version }} sleep infinity
+          # coreutils includes mktemp
+          docker exec test_container apk add coreutils curl
+          if [ "${{ matrix.sed }}" == "GNU" ]; then
+            docker exec test_container apk add sed
+          fi
+          cargo test --all-targets --all-features
+          docker rm --force test_container
       - name: Run clippy check
         run: cargo clippy --all-targets --all-features -- -D warnings
```

Comment:
There is no need to run this several times over matrix. Let separate this into two jobs.